Lib/test/test_socket.py

```diff
@@ -1872,6 +1872,25 @@ def testBCMConstants(self):
         socket.CAN_BCM_RX_TIMEOUT   # cyclic message is absent
         socket.CAN_BCM_RX_CHANGED   # updated CAN frame (detected content change)
 
+        # flags
+        socket.CAN_BCM_SETTIMER
+        socket.CAN_BCM_STARTTIMER
+        socket.CAN_BCM_TX_COUNTEVT
+        socket.CAN_BCM_TX_ANNOUNCE
+        socket.CAN_BCM_TX_CP_CAN_ID
+        socket.CAN_BCM_RX_FILTER_ID
+        socket.CAN_BCM_RX_CHECK_DLC
+        socket.CAN_BCM_RX_NO_AUTOTIMER
+        socket.CAN_BCM_RX_ANNOUNCE_RESUME
+        socket.CAN_BCM_TX_RESET_MULTI_IDX
+        socket.CAN_BCM_RX_RTR_FRAME
+
+    @unittest.skipUnless(hasattr(socket, "CAN_BCM_CAN_FD_FRAME"),
```

Comment:
Is it better to parse the kernel version from `platform.release` as a way of checking whether this exists?

Reply:
Yes, using the `test.support.requires_linux_version` decorator.

Reply:
And don't forget to also use the same decorator as above, i.e.:
```python
    @unittest.skipUnless(hasattr(socket, "CAN_BCM"),
                         'socket.CAN_BCM required for this test.')
    @requires_linux_version(4, 8)
    def testBCMConstantsSupportsFD(self):
        # ...
```

Reply:
Also the method name for the separate test is unclear to me ("supports FD"). Either of `testBCMConstantsLinux48` or `testBCMConstantsFD` seems better.